languages/python/oso/tests/test_data_filtering.py

```diff
@@ -293,6 +319,22 @@ def test_no_relationships(oso, t):
     assert len(results) == 3
 
 
+def test_neq(oso, t):
+    policy = """
+    allow("steve", "get", foo: Foo) if foo.bar.id != "hello";
```

Comment:
What about policies with multiple ne `foo.bar.id != 1 and foo.bar.id != 2`? What about `foo.bar.id = "hello" and foo.bar.id != "goodbye"` or `foo.bar.id = x and foo.bar.id != y`?

Seems like there's more cases we should test here.